src/main/java/com/limechain/sync/warpsync/SyncedState.java

```diff
@@ -180,18 +180,22 @@ public Hash256 loadStateRoot() {
         return storedRootState == null ? null : Hash256.from(storedRootState.toString());
     }
 
-    public void updateSetData(NeighbourMessage neighbourMessage, PeerId peerId) {
-        BigInteger setChangeBlock = neighbourMessage.getLastFinalizedBlock().add(BigInteger.ONE);
+    public synchronized void syncNeighbourMessage(NeighbourMessage neighbourMessage, PeerId peerId) {
```

Comment:
Can we add a comment about the race condition? In the future someone might wonder why this part is synchronized